Rename toTensorListRef -> toTensorVector

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30735 Rename toTensorListRef -> toTensorVector**
* #30734 [jit] remove list specialization from ivalue

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

Differential Revision: [D18814701](https://our.internmc.facebook.com/intern/diff/D18814701)

zdevito committed:
Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

kostmo commented:
## CircleCI build failures summary
As of commit 37abf5b:
* **1/10** broken upstream at merge base d12786b (see [grid view](https://dr.pytorch.org/master-timeline.html?count=10&sha1=d12786b24fc7df0526c9fcd69efa776c53c34e3f&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
* **9/10** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=37abf5b801ff44656a8738b6819528eb564d7a12):

### 10 failures recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775813) pytorch_linux_xenial_py2_7_9_build
**Step:** Build
```
Dec 04 17:02:41 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: could not convert 'element' from 'const at::Dimname' to 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775820) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build
**Step:** Build
```
Dec 04 17:03:08 /pytorch/aten/src/ATen/core/List_inl.h:94:12: error: could not convert ‘element’ from ‘const at::Dimname’ to ‘c10::IValue’
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775825) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build
**Step:** Build
```
Dec 04 17:03:12 /pytorch/aten/src/ATen/core/List_inl.h:94:12: error: could not convert ‘element’ from ‘const at::Dimname’ to ‘c10::IValue’
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775827) caffe2_onnx_py3_6_clang7_ubuntu16_04_build
**Step:** Build
```
Dec 04 17:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: no viable conversion from returned value of type 'const at::Dimname' to function return type 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775834) caffe2_onnx_py2_gcc5_ubuntu16_04_build
**Step:** Build
```
Dec 04 17:03:22 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: could not convert 'element' from 'const at::Dimname' to 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775821) pytorch_linux_xenial_py3_6_gcc5_4_build
**Step:** Build
```
Dec 04 17:03:59 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: could not convert 'element' from 'const at::Dimname' to 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775833) pytorch_linux_xenial_py3_clang5_asan_build
**Step:** Build
```
Dec 04 17:03:56 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: no viable conversion from returned value of type 'const at::Dimname' to function return type 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775823) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build
**Step:** Build
```
Dec 04 17:03:27 FAILED: caffe2/CMakeFiles/torch.dir/operators/resize_3d_op.cc.o
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775814) pytorch_xla_linux_xenial_py3_6_clang7_build
**Step:** Build
```
Dec 04 17:04:33 /var/lib/jenkins/workspace/aten/src/ATen/core/List_inl.h:94:12: error: no viable conversion from returned value of type 'const at::Dimname' to function return type 'c10::IValue'
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3775816) pytorch_macos_10_13_py3_build
**Step:** Build
```
Dec 04 09:05:25 FAILED: caffe2/CMakeFiles/conv_to_nnpack_transform_test.dir/transforms/conv_to_nnpack_transform_test.cc.o
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"

Rename toTensorListRef -> toTensorVector

This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

gh-metadata: pytorch pytorch 30735 gh/zdevito/137/head

zdevito committed:
Update on "Rename toTensorListRef -> toTensorVector"


This makes it more clear that a new copy of the vector is being returned.
A few places bound Ref to ArrayRef which would be invalid in this new world.

Differential Revision: [D18814701](https://our.internmc.facebook.com/intern/diff/D18814701)

[ghstack-poisoned]
